### DISCUSSION TITLE
Persist the same store multiple times depending on user account

### DISCUSSION THREAD

**Discussion by churichard** [pmndrs/zustand#525] [Q&A]
I'm using the zustand persist middleware, but I don't want the state to be shared across user accounts. Is it possible to namespace it? Or in other words, can I use the middleware to persist the same store multiple times?

I've tried to use the user id as part of the name, but I would need some way to recreate the storage whenever a user logs in. Right now it just reuses the same storage as before.

Also, I access the store both inside and outside of React components, so I would need to keep that functionality.

Any help would be appreciated!

**Comment by AnatoleLucet** [comment]
Hi, is this related to #513?

**Reply by churichard** [reply]
Yeah, I think solving that issue would likely solve my problem as well.
